### PR TITLE
[FIX] point_of_sale,pos_sale: fix product info popup

### PR DESCRIPTION
**Pull request by davidmonnom**
Following changes of product card that are now product.template, adapt the product info popup to display the correct information.

Now it display information about the product template instead of the first product variant.

### CODE REVIEW

File: addons/point_of_sale/static/src/app/components/popups/product_configurator_popup/product_configurator_popup.js [hunk 1]
```diff
@@ -174,6 +183,7 @@ export class ProductConfiguratorPopup extends Component {
         }
 
         this.state.product = product;
+        console.log(this.state.product);
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/components/popups/product_configurator_popup/product_configurator_popup.js:186]: remove the console.log

File: addons/pos_sale/models/product_template.py [hunk 2]
```diff
@@ -4,6 +4,17 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
+    def get_product_info_pos(self, price, quantity, pos_config_id):
```
Comment (adgu-odoo) [inline addons/pos_sale/models/product_template.py:7]:
```suggestion
    def get_product_info_pos(self, price, quantity, pos_config_id, product_variant_id=False):
```

File: addons/pos_sale/models/product_template.py [hunk 3]
```diff
@@ -4,6 +4,17 @@
...
+        res = super().get_product_info_pos(price, quantity, pos_config_id)
```
Comment (adgu-odoo) [inline addons/pos_sale/models/product_template.py:8]:
```suggestion
        res = super().get_product_info_pos(price, quantity, pos_config_id, product_variant_id=False)
```

File: addons/pos_sale/models/product_template.py [hunk 4]
```diff
@@ -4,6 +4,17 @@
...
+
+        # Optional products
+        res['optional_products'] = [
+            {'name': p.name, 'price': min(p.product_variant_ids.mapped('lst_price'))}
+            for p in self.optional_product_ids.filtered_domain(self._optional_product_pos_domain())
```
Comment (adgu-odoo) [inline addons/pos_sale/models/product_template.py:13]: `_optional_product_pos_domain` is defined in `product.product` and use in `product.template`, shouldn't you adapt that ?

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_sale/models/product_template.py (hunk 2 above)
Comment (davidmonnom) [reply addons/pos_sale/models/product_template.py:7]: bien vu le sang

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
robodoo r+

**Pull request closed without merging**